### PR TITLE
Add external link examples and improve external link icon display

### PR DESCRIPTION
**Pull request by thisisdano**
This PR adds external link code examples and makes some small stylistic changes to the appearance of the external link icon — making it a little bigger and better aligned with the link text.

### CODE REVIEW

File: src/stylesheets/core/mixins/_external-link.scss [hunk 1]
```diff
@@ -4,15 +4,16 @@
...
     background-image: url("#{$image-path}/#{$external-link}.svg");
-    background-position: 50% 60%;
     background-repeat: no-repeat;
     background-size: 100%;
     content: "";
-    display: inline;
+    display: inline-block;
+    height: $external-link-size;
     margin-left: units(2px);
-    padding-left: $icon-size;
```
Comment (mejiaj) [inline src/stylesheets/core/mixins/_external-link.scss:15]: Removing `inline` and `padding-left` reverts the changes we did to fix the icon breaking into next line [#3395]. 

Testing:
```html
<p>is an alternate external text link on a dark background. is an alternate external text link on a dark background.     is an alternate external text link on a. is an alternate external text link on a dark background.     is an alternate <a class="usa-link usa-link--alt usa-link--external" href="https://i.giphy.com/media/WPzQF6ruiIIVzHNlwX/source.gif">alternate external text link on a dark background</a></p>
```

Produces:
image

Comment (thisisdano) [reply src/stylesheets/core/mixins/_external-link.scss:15]: Ah, thanks — I'll see if I can resolve this

Comment (thisisdano) [reply src/stylesheets/core/mixins/_external-link.scss:15]: I'll leave a comment in the code to explain the reasoning for that technique

Comment (mejiaj) [reply src/stylesheets/core/mixins/_external-link.scss:15]: Sounds good! Resolved

**Review by mejiaj — APPROVED** [review]

**Pull request merged**